services/src/main/java/org/keycloak/services/managers/DefaultBruteForceProtector.java

```diff
@@ -208,17 +177,42 @@ protected void failure(KeycloakSession session, LoginEvent event) {
         logger.debugv("waitSeconds: {0}", waitSeconds);
         logger.debugv("deltaTime: {0}", deltaTime);
 
+        boolean quickWait = false;
         if (waitSeconds == 0) {
             if (last > 0 && deltaTime < realm.getQuickLoginCheckMilliSeconds()) {
                 logger.debugv("quick login, set min wait seconds");
                 waitSeconds = realm.getMinimumQuickLoginWaitSeconds();
+                quickWait = true;
             }
         }
         if (waitSeconds > 0) {
             waitSeconds = Math.min(realm.getMaxFailureWaitSeconds(), waitSeconds);
```

Comment:
Does it make sense to you @douglaspalmer? Indeed, we won't get a quick wait when permanent lockout is enabled ...

Reply:
@pedroigor we always had quick wait with permanent lockout. We didn't use Max Failure Wait Seconds with permanent lockout previously. I can tweak the algorithm so we only use Math.min(realm.getMaxFailureWaitSeconds(), waitSeconds) when temporary lockouts are enabled. Or I can tweak the UI so that maxFailureWaitSeconds is setable for permanent lockout. Which would you prefer?

@gilvansfilho thanks for picking up on this.

Reply:
@douglaspalmer 
Using that `Math.min(realm.getMaxFailureWaitSeconds(), waitSeconds) `for quick login failure will cause side effects, even for temporary lockout.

To be clear: today when occur a quick login failure, the wait time is `realm.getMinimumQuickLoginWaitSeconds()` both for permanent or temporary lockout.

Reply:
Just removed my last comment, what @douglaspalmer said in his last comment is right... :)

Reply:
So today when:

- permanent lockout is ON and occur a quick login failure, wait time are `realm.getMinimumQuickLoginWaitSeconds()`
- permanent lockout is OFF (so temp lockout) and occur a quick login failure, wait time are `Math.min(realm.getMaxFailureWaitSeconds(), realm.getMinimumQuickLoginWaitSeconds())`

what will be behaviour for mixed strategy?